### ISSUE TITLE
Instanceof checks seem to be case sensitive

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#241]
Expected: no error
Observed: `ERROR: InvalidReturnType - src/types.php:5:32 - The declared return type 'MyClass' for convert_object is incorrect, got 'array<mixed, mixed>|MyClass|false'  `

```php
<?php
class myclass {}  // deliberately lowercase for the example

/** @param MyClass|array|false $obj */
function convert_object($obj): MyClass {
    if (!($obj instanceof MyClass)) {
        $obj = new MyClass();
    }   
    return $obj;
}
```

**Comment by muglug** [comment]
Fixed by b7f6d15 – badly-cased classes won't stop further analysis now.

Thanks!
